vulture/core.py

```diff
@@ -461,14 +461,23 @@ def _define(
         confidence=DEFAULT_CONFIDENCE,
         ignore=None,
     ):
+        def _get_first_lineno(first_node, typ):
+            if typ == "property" and sys.version_info < (3, 8):
+                # For python < 3.8, increment as many line numbers as there are
+                # decorators on the method.
+                # From Python3.8 onwards, lineno for property is the same as
+                # that of decorated method.
```

Comment:
Let's turn this into a docstring and rewrite it slightly:

For decorated methods, Python < 3.8 uses the first decorator to report the first line number. To obtain the line number of the actual method definition (as in Python >= 3.8), we add the number of decorators to the line number. Obviously, this is only exact if all decorators use only one line.

Reply:
Can we maybe make this an exact line number (even for multiline decorators)? For this, we would need to be able to retrieve the method's AST node from the property's AST node and call `.lineno` on it. Does the property's AST node have the relevant fields? You can inspect this on Python 2.7 with `dir(node)`.

Reply:
> Obviously, this is only exact if all decorators use only one line.

Yeah, that is a problem. We need more test cases here.

> For this, we would need to be able to retrieve the method's AST node from the property's AST node and call .lineno on it

I'm a little confused with this. Aren't AST nodes for property and method the same?

Reply:
What I meant was the FunctionDef AST node and a *decorator* AST node. To retrieve the method line number from the FunctionDef node `node`, we could use `lines.get_last_line_number(node.decorator_list[-1])`. However, this is also just an approximation. Thinking about this again, I belive that it would be best to simply report the first line number of the first decorator for unused properties for Python 3.8+. This preserves the behaviour of Vulture for newer Python versions and is also more accurate for counting the size of the unused code chunk (if the property is unused, we also don't need it's decorators). Most importantly, it's easy to compute the line number accurately (should work for all Python versions, but deserves a comment about why we don't use `node.lineno`): `node.decorator_list[0].lineno`.

Reply:
I tried `lines.get_last_line_number(node.decorator_list[-1])` for a bunch of cases, and I agree about the inaccuracy, eg., it would fail here:

```python
class TestFoo:
    @pytest.mark.parametrize(
        "line, codes",
        [
            ("# noqa", ["all"]),
            ("## noqa", ["all"]),
        ],
    )
    @property
    @pytest.mark.parametrize(
        "line, codes",
        [
            ("# noqa", ["all"]),
        ],
    )
    def test_noqa_regex_present(line, codes):
        pass
```

Also, I can't remember why this PR only targeted properties, and not all decorated functions? 😅

Anyways, I'll update this PR for all functions with the changes you mentioned above.